### PR TITLE
Add php-cs-fixer formatter

### PR DESCRIPTION
**Pull request by shurizzle**
Add php-cs-fixer formatter as a formatter from github. I haven't used composer because it has no deps and they distribute it as a single file in releases.

### CODE REVIEW

File: lua/mason-registry/php-cs-fixer/init.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
+local Pkg = require "mason-core.package"
+local _ = require "mason-core.functional"
+local github = require "mason-core.managers.github"
+local std = require "mason-core.managers.std"
+
+return Pkg.new {
+    name = "php-cs-fixer",
+    desc = _.dedent [[
+        Portable package manager for Neovim that runs everywhere Neovim runs. Easily install and manage LSP servers, DAP
+        servers, linters, and formatters.
```
Comment (williamboman) [inline lua/mason-registry/php-cs-fixer/init.lua:10]:
```suggestion
        The PHP Coding Standards Fixer (PHP CS Fixer) tool fixes your code to follow standards; whether you want to
        follow PHP coding standards as defined in the PSR-1, PSR-2, etc., or other community driven ones like the
        Symfony one. You can also define your (team's) style through configuration.')
```

**Review by williamboman — COMMENTED** [review]
Will need to check this on Windows, unless perhaps you've verified it works there?

**Comment by williamboman** [comment]
@williambotman /fixup

**Comment by shurizzle** [comment]
> Will need to check this on Windows, unless perhaps you've verified it works there?

No and I do not have a Windows machine configured to test it, sorry

**Comment by williamboman** [comment]
> No and I do not have a Windows machine configured to test it, sorry

No worries! I really need to get a multi platform build system hooked up to CI test suite soon..

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/lua/mason-registry/php-cs-fixer/init.lua b/lua/mason-registry/php-cs-fixer/init.lua
index f64698b..0929a2b 100644
--- a/lua/mason-registry/php-cs-fixer/init.lua
+++ b/lua/mason-registry/php-cs-fixer/init.lua
@@ -2,6 +2,8 @@ local Pkg = require "mason-core.package"
 local _ = require "mason-core.functional"
 local github = require "mason-core.managers.github"
 local std = require "mason-core.managers.std"
+local platform = require "mason-core.platform"
+local path = require "mason-core.path"

 return Pkg.new {
     name = "php-cs-fixer",
@@ -20,10 +22,26 @@ return Pkg.new {
             .download_release_file({
                 repo = "FriendsOfPHP/PHP-CS-Fixer",
                 asset_file = "php-cs-fixer.phar",
-                out_file = "php-cs-fixer",
+                out_file = platform.is.win and "php-cs-fixer.phar" or "php-cs-fixer",
             })
             .with_receipt()
-        std.chmod("+x", { "php-cs-fixer" })
-        ctx:link_bin("php-cs-fixer", "php-cs-fixer")
+        platform.when {
+            unix = function()
+                std.chmod("+x", { "php-cs-fixer" })
+                ctx:link_bin("php-cs-fixer", "php-cs-fixer")
+            end,
+            win = function()
+                ctx:link_bin(
+                    "php-cs-fixer",
+                    ctx:write_shell_exec_wrapper(
+                        "php-cs-fixer",
+                        ("php %q"):format(path.concat {
+                            ctx.package:get_install_path(),
+                            "php-cs-fixer.phar",
+                        })
+                    )
+                )
+            end,
+        }
     end,
 }
```

**Pull request merged**